src/Equinox.Core/Cache.fs

```diff
@@ -34,11 +34,13 @@ type private CacheEntry<'state>(initialToken: StreamToken, initialState: 'state,
     /// Coordinates having a max of one in-flight request across all staleness-tolerant loads at all times
     // Follows high level flow of AsyncCacheCell.Await - read the comments there, and the AsyncCacheCell tests first!
     member x.ReadThrough(maxAge: TimeSpan, isStale, load: Func<_, _>) = task {
+        let act = System.Diagnostics.Activity.Current
         let cacheEntryValidityCheckTimestamp = System.Diagnostics.Stopwatch.GetTimestamp()
         let isWithinMaxAge cachedValueTimestamp = Stopwatch.TicksToSeconds(cacheEntryValidityCheckTimestamp - cachedValueTimestamp) <= maxAge.TotalSeconds
         let fetchStateConsistently () = struct (cell, tryGet (), isWithinMaxAge verifiedTimestamp)
         match lock x fetchStateConsistently with
         | _, ValueSome cachedValue, true ->
+            if act <> null then act.AddCacheHit(true)
```

Comment:
this cache hit is different to the other one, which is only using the cache entry as a base for the load (and this one has a max age associated, see the .AddStale comment

Reply:
Also on L55, could record whether we waited for someone else to shoot the bear, or we were the instigator (not bothered as much about that when compared to the "used cached value directly" vs "used cache entry to do an incremental load" fact)